### ISSUE TITLE
Connector dependency task failed on DBT update PR

### ISSUE THREAD

**Issue by grishick** [airbytehq/airbyte#16650]

**Comment by etsybaev** [comment]
https://github.com/airbytehq/airbyte/pull/16339

**Comment by etsybaev** [comment]
Hi @grishick .
Is there any way to reproduce it locally?
Thanks

**Comment by grishick** [comment]
I don't think so, please talk to @suhomud who authored the initial version of the GitHub action

**Comment by etsybaev** [comment]
NOTES:
Basically the issue occurs for only PR from forked repos as "git diff" doesn't work for such cases.

API that may help to solve the issue, but it's not that straight-forward.
https://docs.github.com/en/rest/commits/commits#compare-two-commits
EX. GET https://api.github.com/repos/airbytehq/airbyte/compare/master...etsybaev:etsybaev-test-fork-branch?per_page=1&page=2

But there is some issue with pagination here
https://github.com/orgs/community/discussions/33505

**Comment by etsybaev** [comment]
Performed scoping.
Basically the issue occurs for only PR from forked repos as "git diff" doesn't work for such cases.

There is an API that potentially could solve the issue:
https://docs.github.com/en/rest/commits/commits#compare-two-commits
EX. GET https://api.github.com/repos/airbytehq/airbyte/compare/master...etsybaev:etsybaev-test-fork-branch?per_page=1&page=2

Once github will fix the issue with pagination we have replaced this "git giff"  call with some python script. I created some draft, see attached file.


But there is some issue with pagination. Created an issue for them, but there are no replies for now.
https://github.com/orgs/community/discussions/33505


Another challenge would be to find the customer's repo name in GitHub Action's envs in the runtime. For now it seems like the "github.actor" is the closest one suitable for us https://docs.github.com/en/actions/learn-github-actions/contexts
I.e. if PR comes from Airbyte repo - then insert just a branch name, if PR comes from forked repo - then also use "epo_name:branch_name" (ex.)

**Comment by etsybaev** [comment]
import json
import sys
import requests

df=open('changed_files_new.txt','w')

page = 1

while True:
    print(f"page is {page}!")

    response = requests.get('https://api.github.com/repos/airbytehq/airbyte/compare/master...' + sys.argv[1] + '?per_page=30&page=' + str(page))
    jsonResponse = response.json()
    responseAsJson = json.loads(response.text)
    files = responseAsJson.get("files")

    if files is None:
        print('variable is not null')
        break;

    for file in files:
        if file['filename'].startswith("airbyte-integrations/"):
            print(file['filename'])
            df.write(file['filename'])
            df.write('\n')

    page = page + 1

df.close()
